staging/src/k8s.io/apiserver/pkg/endpoints/handlers/workflow.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handlers
+
+import (
+	"net/http"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// guessWorkflowID tries to find a reasonable default for the
+// workflowID, based on user-agent of the incoming request,
+// sub-resource, verb and content-type. Always returns a non-empty
+// string.
+func guessWorkflowID(userAgent, subResource, verb, contentType string) string {
+	// This is the worst-case scenario: we don't have anything useful
+	workflowID := "default"
```

Comment:
Is this useful if it's a prefix for literally everything?

Reply:
It's not, this is pretty much the default if no userAgent is found.